source/concurrent/Parallel.h

```diff
@@ -0,0 +1,157 @@
+/* Parallel.h
+Copyright (c) 2024 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef PARALLEL_H_
+#define PARALLEL_H_
+
+#include "Scheduler.h"
+
+#include <version>
+
+
+
+// Only use this header if the underlying system doesn't support the standard algorithms,
```

Comment:
Are we developing a game, or a multi-threading library? Using the default c++17 libraries sounds relatively safe. And falling back to standard single-threaded operations for non-compliant systems sounds relatively safe as well.
But just implementing our own multi-threading library because one of the platforms we run on is lagging behind in c++17 support sounds a bit out of scope for the ES project.

Reply:
It's not an issue caused by older platforms; if we compile the game, it should work just fine on older systems.

The standard library is rather vague on the implementation of parallel functions, so these are different between the standard libraries provided by each compiler. Luckily, these differences are minimal, since both GCC and Clang piggybacks on the work Intel did for its parallel stl. This is done using a library called Intel Threading Building Blocks (TBB).

While TBB is supported on all major operating systems, it seems there is a bug in the current version of the vcpkg port of TBB that prevents building it in our setup under Windows (mingw).

TL;DR A third-party bug prevents us from using the standard functions on windows, so I provided a barebones implementation to get some of the benefits until it's fixed.

(There was also an issue with macos not following the standard, I think that has some extra workaround code here, but I don't recall the details right now.)

Reply:
I still prefer that we just use the versions that are working well, and don't provide any fixes for implementations with bugs or missing functions, but just run single-threaded on those buggy or incomplete platforms.

If the bugfixes are expected soon, when we could also just wait until they are available and then switch to the fixed versions.

Reply:
It's just a matter of vcpkg picking up the latest tbb version. I certainly hope it happens soon.

Yeah, I would prefer to not have the workarounds here as well. They are necessary if I want to test this effectively before the vcpkg fix, but I would be surprised if they would still be necessary by the next unstable cycle.